src/platform_impl/macos/app_delegate.rs

```diff
@@ -79,3 +106,45 @@ extern "C" fn did_resign_active(this: &Object, _: Sel, _: id) {
     }
     trace!("Completed `applicationDidResignActive`");
 }
+
+extern "C" fn application_open_file(_this: &Object, _: Sel, _sender: id, filename: id) -> BOOL {
+    trace!("Triggered `application:openFile:`");
+    let mut filenames_vec = Vec::with_capacity(1);
+    let string = ns_string_to_rust(filename);
+    filenames_vec.push(string.into());
+    let event = Event::OpenFiles(filenames_vec);
+    AppState::queue_event(EventWrapper::StaticEvent(event));
+    trace!("Completed `application:openFile:`");
+
+    // Return true to indicate to the OS that the file type is supported.
+    // (If the filetype turns out not to be supported, it's the application's
+    // responsibility to inform the user)
```

Comment:
Why is this singling out the filetype? There are so many other reasons why I file could potentially not be opened. It might be gone, the user might not have permissions to open it, or someone else might already be busy with it.

This is also not documentation for the user, this is internal documentation. So we should document here *why* we chose to always return `YES` instead of actually giving it a proper value.